packages/report-server/src/reportBuilder/output/output.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { Row } from '../types';
+import { outputBuilders } from './functions/outputBuilders';
+
+type OutputParams = {
+  type: string;
+  config: unknown;
+};
+
+const output = (rows: Row[], params: OutputParams) => {
+  const { type, config } = params;
+
+  const outputBuilder = outputBuilders[type as keyof typeof outputBuilders](config);
+  return outputBuilder(rows);
+};
+
+const buildParams = (params: unknown): OutputParams => {
+  const defaultType = { type: 'default', config: undefined };
+
+  if (typeof params === 'undefined') {
+    return defaultType;
+  }
+  if (typeof params === 'object') {
+    if ('matrix' in params) {
```

Comment:
I think once the config is re-worked into being a flat object, it'd be good to push all the outputType specific validation logic into their own `buildParams()` function 👍

Reply:
Yeah I re-work them and it looks so clean now 👍